lib/options.nix

```diff
@@ -46,8 +46,10 @@ rec {
     internal ? null,
     # Whether the option shows up in the manual.
     visible ? null,
-    # Whether the option can be set only once
+    # Whether the option can be set only once.
     readOnly ? null,
+    # Anything extra.
+    extra ? null
```

Comment:
This is an unnecessarily hacky way to get the functionality you're implementing with it. If you need additional module settings when options have a certain value, use a proper module for that instead of abusing mkOption's arguments.

Do to this you'll need to do a data transformation for `config.nix`, changing all the `foo = mkUse { ... }` into a list of modules that each define the option and do the extra things.

Or just write it as a normal module, with additional config being applied in the config section.